projects/ccf-rui/src/app/shared/components/rotation-slider/rotation-slider.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+import { Rotation } from '../../../core/models/rotation';
+
+/**
+ * Component that enables the setting of a Rotation object via either 3 draggable sliders
+ * or through an Input method.
+ */
+@Component({
+  selector: 'ccf-rotation-slider',
+  templateUrl: './rotation-slider.component.html',
+  styleUrls: ['./rotation-slider.component.scss']
+})
+export class RotationSliderComponent {
+  /**
+   * Input that allows the rotation to be changed from outside of the component
+   */
+  @Input() rotation: Rotation = { x: 0, y: 0, z: 0 };
+
+  /**
+   * Output that emits the new rotation whenever it is changed from within the component
+   */
+  @Output() rotationChanged = new EventEmitter<Rotation>();
+
+  /**
+   * Function that handles updating the rotation and emitting the new value
+   * @param newRotation the new value for one of the axis to be set to
+   * @param axis which axis to update
+   */
+  changeRotation(newRotation: number | string, axis: string): void {
+    if (typeof (newRotation) === 'string') {
+      newRotation = parseInt(newRotation, 10);
+    }
+
+    this.rotation[axis] = newRotation;
```

Comment:
Create a new object instead of modifying the current one. Otherwise we might end up with change detection bugs